go/parquet/pqarrow/path_builder.go

```diff
@@ -440,7 +440,9 @@ func (p *pathBuilder) Visit(arr arrow.Array) error {
 		}
 		return nil
 	case arrow.EXTENSION:
-		return xerrors.New("extension types not implemented yet")
+		// TODO: (hermanschaaf) handle nested extension types
```

Comment:
it should be feasible to just do `return p.Visit(arr.(arrow.ExtensionArray).Storage())` and that would handle nested extension types for you just fine.

Reply:
The test is finally added